examples/derived_gauges/derived_gauge.go

```diff
@@ -0,0 +1,237 @@
+	defer q.mu.Unlock()
+	return time.Now().Sub(q.lastConsumed).Seconds()
+}
+
+// END tofloat64
+
+func getInput() int {
+	reader := bufio.NewReader(os.Stdin)
+	limit := 100
+	for {
+		fmt.Printf("Enter number of items to put in consumer queue? [1-%d]: ", limit)
+		text, _ := reader.ReadString('\n')
+		count, err := strconv.Atoi(strings.TrimSuffix(text, "\n"))
+		if err == nil {
+			if count < 1 || count > limit {
+				fmt.Printf("invalid value %s\n", text)
+				continue
+			}
+			return count
+		}
+		fmt.Printf("error %v\n", err)
+	}
+}
+
+func doWork() {
+	fmt.Printf("Program monitors queue using two derived gauge metrics.\n")
+	fmt.Printf("   1. queue_size = the instantaneous size of the queue.\n")
+	fmt.Printf("   2. queue_seconds_since_processed_last = the number of seconds elapsed since last time the queue was processed.\n")
+	fmt.Printf("Go to http://localhost:9090/metrics to see the metrics.\n\n\n")
+
+	// Take a number of items to queue as an input from the user
+	// and enqueue the same number of items on to the consumer queue.
+	for {
+		count := getInput()
+		q.produce(count)
+		fmt.Printf("press CTRL+C to terminate the program\n")
+	}
+}
+
+func createAndStartExporter() {
+	// Create Prometheus metrics exporter to verify derived gauge metrics in this example.
+	exporter, err := prometheus.NewExporter(prometheus.Options{})
+	if err != nil {
+		log.Fatalf("Failed to create the prometheus metrics exporter: %v", err)
+	}
+	http.Handle("/metrics", exporter)
+	go func() {
+		log.Fatal(http.ListenAndServe(":9090", nil))
+
+	}()
+}
+
+func main() {
+	createAndStartExporter()
+
+	// Create metric registry and register it with global producer manager.
+	// START reg
+	r := metric.NewRegistry()
+	metricproducer.GlobalManager().AddProducer(r)
+	// END reg
+
+	// Create Int64DerviedGauge
+	// START size
+	queueSizeGauge, err := r.AddInt64DerivedGauge(
+		"queue_size",
+		metric.WithDescription("Instantaneous queue size"),
+		metric.WithUnit(metricdata.UnitDimensionless))
+	if err != nil {
+		log.Fatalf("error creating queue size derived gauge, error %v\n", err)
+	}
+	// END size
+
+	// START entrySize
+	err = queueSizeGauge.UpsertEntry(q.Size)
+	if err != nil {
+		log.Fatalf("error getting queue size derived gauge entry, error %v\n", err)
+	}
+	// END entrySize
+
+	// Create Float64DerviedGauge
+	// START elapsed
+	elapsedSeconds, err := r.AddFloat64DerivedGauge(
+		"queue_seconds_since_processed_last",
+		metric.WithDescription("time elapsed since last time the queue was processed"),
+		metric.WithUnit(metricdata.UnitDimensionless))
+	if err != nil {
+		log.Fatalf("error creating queue_seconds_since_processed_last derived gauge, error %v\n", err)
+	}
+	// END elapsed
+
+	// START entryElapsed
+	err = elapsedSeconds.UpsertEntry(q.Elapsed)
+	if err != nil {
+		log.Fatalf("error getting queue_seconds_since_processed_last derived gauge entry, error %v\n", err)
+	}
+	// END entryElapsed
+
+	cQuit := make(chan bool)
+	defer func() {
+		cQuit <- true
```

Comment:
You don't need to send true, if you close channel that will be sufficient for the select.

Reply:
done.